streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -1088,7 +1094,11 @@ private long computeLatency() {
      */
     public void shutdown() {
         log.info("Informed to shut down");
-        setState(State.PENDING_SHUTDOWN);
+        State oldState = setState(State.PENDING_SHUTDOWN);
+        if (oldState == State.CREATED) {
+            // The thread may not have been started. Take responsibility for shutting down
```

Comment:
nit: `the thread never started, thus need to take responsibility for shutting down`

Reply:
It may have started but not been scheduled yet.